### PR TITLE
Update various dependencies

### PR DESCRIPTION
**Pull request by wendigo**
No release notes needed.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -516,13 +517,13 @@
...
                 <artifactId>h2</artifactId>
-                <version>2.2.220</version>
+                <version>2.2.224</version>
             </dependency>
 
             <dependency>
                 <groupId>com.microsoft.sqlserver</groupId>
                 <artifactId>mssql-jdbc</artifactId>
-                <version>12.2.0.jre11</version>
+                <version>12.4.1.jre11</version>
```
Comment (mosabua) [inline pom.xml:526]: We should switch to the jre 17 version...

**Comment by mosabua** [comment]
You probably need to verify for each update that no release notes entry is needed. For example, for the MS JDBC driver we did need release notes entries in the past in some cases..

File: pom.xml (hunk 1 above)
Comment (wendigo) [reply pom.xml:526]: There is no such build :)

Comment (mosabua) [reply pom.xml:526]: oh wow .. when did they switch that off and remove it again. I missed that.

Comment (mosabua) [reply pom.xml:526]: Turns out they dropped it in the 12.x series .. https://search.maven.org/artifact/com.microsoft.sqlserver/mssql-jdbc

File: lib/trino-filesystem/pom.xml [hunk 2]
```diff
@@ -38,7 +38,7 @@
         </dependency>
 
         <dependency>
-            <groupId>io.opentelemetry</groupId>
+            <groupId>io.opentelemetry.semconv</groupId>
```
Comment (ebyhr) [inline lib/trino-filesystem/pom.xml:41]: This commit can be dropped.

**Review by ebyhr — APPROVED** [review]

File: lib/trino-filesystem/pom.xml (hunk 2 above)
Comment (wendigo) [reply lib/trino-filesystem/pom.xml:41]: Yeah, I'm dropping it prior to the merge.

**Pull request merged**